core/ir/opnd_api.h

```diff
@@ -1766,13 +1763,8 @@ typedef enum {
 #    define DR_REG_STOP_DR DR_REG_DR15    /**< End of debug register enum values */
```

Comment:
What about DR_REG_VIRT* for regdeps?  The DR_REG_VIRT* are orthogonal to the main arch DR_REG_ enum?  They are guaranteed to have fewer values then the enum no matter the arch?  So no change here is needed.  Right? @edeiana to confirm.

Reply:
Yes, they are guaranteed to have fewer values than the DR_REG_ enum of the architecture where the trace was collected.
However, it shouldn't matter, since we're changing the ISA to DR_ISA_REGDEPS.
One could collect a trace on AARCH64, convert it to DR_ISA_REGDEPS, and analyze it on x86.
We explicitly say that DR_REG_VIRT* are just numeric values to track register dependencies, they are not real registers.
We do handle `reg_get_size()` because we set `opnd->size` to be the same value returned by `get_operation_size()`, but other register operations (e.g., `opnd_create_far_base_disp_ex()`) are undefined because they don't really make sense on a REGDEPS trace (for the reason above).

Reply:
My question is whether we need to add DR_REG_AFTER_LAST_VALID_ENUM at the end of the DR_REG_VIRT* list: but since the build's arch will already have that defined for the arch-specific enum list I suppose we do nothing and let all the checks and uses of the last enum use the arch-specific value.

Reply:
We don't need to add it.
`DR_REG_AFTER_LAST_VALID_ENUM` will already be mapped to one `DR_REG_VIRT`.